### PR TITLE
Run the aux gen protocol when resharing to a new signing committee

### PR DESCRIPTION
**Pull request by ameba23**
Closes https://github.com/entropyxyz/entropy-core/issues/976

When running the reshare protocol, we also need to re-generate aux info and store the aux info together with the new keyshare.

When running the refresh protocol (which is just a reshare with the parties not changing), we need to store the old aux info together with the new keyshare - as both are stored under the same key in the key-value db.

### CODE REVIEW

File: crates/protocol/src/lib.rs [hunk 1]
```diff
@@ -214,9 +214,9 @@ impl SessionId {
     }
 }
 
-/// A sub-protocol of the DKG protocol
+/// A sub-protocol of the DKG or reshare protocols
 #[derive(Debug)]
-pub enum DkgSubsession {
+pub enum Subsession {
```
Comment (ameba23) [inline crates/protocol/src/lib.rs:219]: I renamed this enum because it is no longer only used for DKG

File: crates/testing-utils/Cargo.toml [hunk 2]
```diff
@@ -20,7 +20,7 @@ tokio             ={ version="1.39", features=["macros", "fs", "rt-multi-thread"
 axum              ={ version="0.7.5" }
 entropy-shared    ={ version="0.2.0", path="../shared" }
 entropy-kvdb      ={ version="0.2.0", path="../kvdb", default-features=false }
-entropy-tss       ={ version="0.2.0", path="../threshold-signature-server" }
+entropy-tss       ={ version="0.2.0", path="../threshold-signature-server", features = ["test_helpers"] }
```
Comment (ameba23) [inline crates/testing-utils/Cargo.toml:23]: Unrelated to this PR but i needed this for testing locally

**Review by JesseAbram — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -22,6 +22,7 @@ At the moment this project **does not** adhere to
 - Reshare confirmation ([#965](https://github.com/entropyxyz/entropy-core/pull/965))
 - Set inital signers ([#971](https://github.com/entropyxyz/entropy-core/pull/971))
 - Add parent key threshold dynamically ([#974](https://github.com/entropyxyz/entropy-core/pull/974))
+- Run the aux gen protocol when resharing to a new signing committee ([#977](https://github.com/entropyxyz/entropy-core/pull/977))
```
Comment (HCastano) [inline CHANGELOG.md:25]: This is more of an internal change, don't think we need to have this

File: crates/threshold-signature-server/src/signing_client/api.rs [hunk 4]
```diff
@@ -112,7 +112,18 @@ pub async fn proactive_refresh(
                 ocw_data.block_number,
             )
             .await?;
-            let serialized_key_share = key_serialize(&new_key_share)
+
+            // Get aux info from existing entry
+            let aux_info = {
+                let existing_entry = app_state.kv_store.kv().get(&key).await?;
+                let (_old_key_share, aux_info): KeyShareWithAuxInfo = deserialize(&existing_entry)
```
Comment (HCastano) [inline crates/threshold-signature-server/src/signing_client/api.rs:119]: I would add a comment here indicating that we intentionally want to use the old `aux_info` in the case of a refresh

**Review by HCastano — APPROVED** [review]
Good stuff 👍

**Pull request merged**